Lib/email/utils.py

```diff
@@ -351,3 +324,11 @@ def localtime(dt=None, isdst=None):
     if dt is None:
         dt = datetime.datetime.now()
     return dt.astimezone()
+
+
+def __getattr__(attr):
+    # lazy import, to speed up module import time
+    if attr == "make_msgid":
+        from email._msgid import make_msgid
+        return make_msgid
+    raise AttributeError(f"module {__name__!r} has no attribute {attr!r}")
```

Comment:
In general I don't like performance hacks like this, but they do have their place.  I can't speak to whether or not this is a worthwhile performance hack, you should seek approval from the maintainers of the impacted modules for that.

That said, the stdlib itself makes no use of make_msgid, and email.utils is not itself considered part of the new email API.  Moving it into a separate module and making that part of the non-legacy public API of the email module would actually make some sense.  I guess we'd just call it 'msgid'?  Then this code should issue a deprecation warning pointing to the new way to import make_msgid.  It feels kind of weird to have a module with just one function, but there isn't really anything else related to it that I can think of.  (I wonder...maybe make_msgid actually belongs in the UUID module?  Probably not.  Wrong RFC.)